src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/SymbolKey/SymbolKey.EventSymbolKey.cs

```diff
@@ -14,22 +14,30 @@ public sealed override void Create(IEventSymbol symbol, SymbolKeyWriter visitor)
         {
             visitor.WriteString(symbol.MetadataName);
             visitor.WriteSymbolKey(symbol.ContainingType);
+            visitor.WriteBoolean(symbol.PartialDefinitionPart is not null);
         }
 
         protected sealed override SymbolKeyResolution Resolve(
             SymbolKeyReader reader, IEventSymbol? contextualSymbol, out string? failureReason)
         {
             var metadataName = reader.ReadString();
             var containingTypeResolution = reader.ReadSymbolKey(contextualSymbol?.ContainingType, out var containingTypeFailureReason);
+            var isPartialImplementationPart = reader.ReadBoolean();
 
             if (containingTypeFailureReason != null)
             {
                 failureReason = $"({nameof(EventSymbolKey)} {nameof(containingTypeResolution)} failed -> {containingTypeFailureReason})";
                 return default;
             }
 
-            using var result = GetMembersOfNamedType<IEventSymbol>(containingTypeResolution, metadataName);
-            return CreateResolution(result, $"({nameof(EventSymbolKey)} '{metadataName}' not found)", out failureReason);
+            using var events = GetMembersOfNamedType<IEventSymbol>(containingTypeResolution, metadataName);
+
+            if (events.Count == 1 && isPartialImplementationPart)
```

Comment:
It looks like this is making use of an assumption that events are not overloaded. Is this really what we want to do when, say, erroneous duplicated partial events are present in source?